robocorp-code/vscode-client/src/viewsWorkItems.ts

```diff
@@ -0,0 +1,203 @@
+import * as vscode from 'vscode';
+import { resolve, join, dirname, basename } from 'path';
+
+import { logError } from './channel';
+import { ROBOCORP_LIST_WORK_ITEMS_INTERNAL } from "./robocorpCommands";
+import { FSEntry, RobotEntry, treeViewIdToTreeView } from "./viewsCommon";
+import { TREE_VIEW_ROBOCORP_ROBOTS_TREE, TREE_VIEW_ROBOCORP_WORK_ITEMS_TREE } from './robocorpViews';
+import { getCurrRobotDir, RobotSelectionTreeDataProviderBase } from './viewsRobotSelection';
+
+const WORK_ITEM_TEMPLATE = `[
+  {
+      "payload": {
+         "message": "Hello World!"
+      },
+      "files": {
+          "orders.xlsx": "orders.xlsx"
+      }
+  }
+]`;
+
+async function getWorkItemInfo(): Promise<WorkItemsInfo | null> {
+  // Would there be a faster way of getting the work item path other than querying the work item info?
+  // The work item tree provider does have it already, but couldn't access it through existing interface.
+  // Would also prefer the keep the static path definition only at the server side.
+  const currTreeDir: FSEntry | undefined = await getCurrRobotDir();
+  const workItemsResult: ActionResultWorkItems = await vscode.commands.executeCommand(ROBOCORP_LIST_WORK_ITEMS_INTERNAL, { robot: resolve(currTreeDir.filePath) });
+  if (!workItemsResult.success) {
+      return;
+  }
+  return workItemsResult.result;
+}
+
+async function queryNewWorkItemName(): Promise<string> {
+  const filename: string = await vscode.window.showInputBox({
+      'prompt': 'Please provide work item name',
+      'ignoreFocusOut': true,
+  });
+  return filename;
+}
+
+async function createNewWorkItem(workItemInfo: WorkItemsInfo, workItemName: string): Promise<void> {
+  if (!workItemInfo?.input_folder_path || !workItemName) {
+      return;
+  }
+
+  const targetFolder = join(workItemInfo.input_folder_path, workItemName);
+  const targetFile = join(targetFolder, 'work-items.json');
+  try {
+      await vscode.workspace.fs.createDirectory(vscode.Uri.file(targetFolder));
+      await vscode.workspace.fs.writeFile(vscode.Uri.file(targetFile), Buffer.from(WORK_ITEM_TEMPLATE));
+      vscode.window.showTextDocument(vscode.Uri.file(targetFile));
+  } catch (err) {
+      logError('Unable to create file.', err);
+      vscode.window.showErrorMessage('Unable to create file. Error: ' + err.message);
+  }
+}
+
+export async function newWorkItemInWorkItemsTree(): Promise<void> {
+  const workItemInfo = await getWorkItemInfo();
+  const workItemName = await queryNewWorkItemName();
+  await createNewWorkItem(workItemInfo, workItemName);
+}
+
+export async function deleteWorkItemInWorkItemsTree(): Promise<void> {
+  let robotContentTree = treeViewIdToTreeView.get(TREE_VIEW_ROBOCORP_WORK_ITEMS_TREE);
+  if (!robotContentTree) {
+      return undefined;
```

Comment:
`await vscode.window.showInformationMessage("No Robot selected for work item selected for deletion.")`